File: ts-multi-region-s3-crr-kms-cmk/multi-region-s3-crr-kms-cmk/lib/multi-region-s3-crr-kms-cmk-target-stack.ts
import { Construct } from 'constructs';
import * as cdk from 'aws-cdk-lib';
import * as s3 from 'aws-cdk-lib/aws-s3';
import { MultiRegionS3CrrKmsCmkTarget } from 'multi-region-s3-crr-kms-cmk-target';

export class MultiRegionS3CrrKmsCmkTargetStack extends cdk.Stack {
    public targetBucket: s3.Bucket;
    public targetKeyIdSsmParameterName: string;

    constructor(scope: Construct, id: string, props?: cdk.StackProps) {
        super(scope, id, props);

        const myTargetConstruct = new MultiRegionS3CrrKmsCmkTarget(this, 'MyTarget');
        this.targetBucket = myTargetConstruct.targetBucket;
        this.targetKeyIdSsmParameterName = myTargetConstruct.targetKeyIdSsmParameterName;
    }
}

File: go-multi-region-s3-crr-kms-cmk/multi-region-s3-crr-kms-cmk/internal/stacks/multi-region-s3-crr-kms-cmk-target-stack.go
package stacks

import (
	"multi-region-s3-crr-kms-cmk/pkg/target"

	"github.com/aws/aws-cdk-go/awscdk/v2"
	"github.com/aws/aws-cdk-go/awscdk/v2/awss3"
	"github.com/aws/constructs-go/constructs/v10"
	"github.com/aws/jsii-runtime-go"
)

type MultiRegionS3CrrKmsCmkTargetStack struct {
	Stack                       awscdk.Stack
	TargetBucket                awss3.Bucket
	TargetKeyIdSsmParameterName string
	Region                      *string
}

func NewMultiRegionS3CrrKmsCmkTargetStack(scope constructs.Construct, id *string, props *awscdk.StackProps) MultiRegionS3CrrKmsCmkTargetStack {
	stack := awscdk.NewStack(scope, id, props)

	myTargetConstruct := target.NewMultiRegionS3CrrKmsCmkTarget(stack, jsii.String("MyTarget"))

	return MultiRegionS3CrrKmsCmkTargetStack{
		Stack:                       stack,
		TargetBucket:                myTargetConstruct.TargetBucket,
		TargetKeyIdSsmParameterName: myTargetConstruct.TargetKeyIdSsmParameterName,
		Region:                      stack.Region(),
	}
}
